### PR TITLE
[google_maps_flutter] Pin GoogleMaps cocoapod dep to last good known version

### PR DESCRIPTION
**Pull request by ditman**
## Description

GoogleMaps 3.10 seems to have increased the minimum requirement for XCode beyond what our test infra can handle (undocumented?).

This change pins the GoogleMaps podspec to <3.10 until the situation clears, to unblock the build trees of both flutter/plugins and flutter/flutter.

**Why are our apps resolving to 3.10, when our Podspec states that ios 8.0 needs to be supported?** 

Apps were resolving to 3.10 because `flutter create` now creates apps that have a minimum iOS version of 9.0 (instead of 8.0).

**Why did 3.10 break us?**

3.10 breaks us because they're using some APIs only available in XCode 11.4 and our infra is in 11.3.1. (However, as @jmagman pointed out, they should check for the availability of these APIs or a runtime error will happen in devices running iOS < 13.4.)

## Related Issues

* https://github.com/flutter/flutter/issues/63447
* b/163359804
* b/163474612

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.5.29+1
+
+* (ios) Pin dependency on GoogleMaps pod to `~>2.7`, to address https://github.com/flutter/flutter/issues/63447
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:3]: Looks like you are pinning on `3.9`, right?

**Comment by jmagman** [comment]
```
    [!] CocoaPods could not find compatible versions for pod "GoogleMaps":
      In Podfile:
        google_maps_flutter (from `.symlinks/plugins/google_maps_flutter/ios`) was resolved to 0.0.2, which depends on
          GoogleMaps (= 3.9)

    Specs satisfying the `GoogleMaps (= 3.9)` dependency were found, but they required a higher minimum deployment target.
```
This is what I saw locally!  Why is it ever resolving to 3.10...

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md (hunk 1 above)
Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:3]: Haha yep

**Comment by ditman** [comment]
@jmagman maybe other plugins are requiring a min version of the SDK > 8, and when using them all together it picks the highest number? Let me audit *other* podspecs.

**Comment by ditman** [comment]
I've looked at an [old build](https://cirrus-ci.com/task/6072739851141120?command=build#L121-L122), and it resolved to:

```
Google Maps SDK for iOS version: 2.7.30514.0
New version of Google Maps SDK for iOS available: 3.9.0.0
```

I officially don't understand how we are getting 3.10 :/ Maybe 3.10 declares that it supports iOS 8?

**Comment by ditman** [comment]
Turns out the `build_all_plugins_ipa` uses `flutter create` to [auto-generate an app](https://github.com/flutter/plugin_tools/blob/master/lib/src/create_all_plugins_app_command.dart#L52-L62) with all the plugins.

The minimum version of ios was bumped from 8.0 to 9.0 [recently](https://github.com/flutter/flutter/pull/62902).

The plugin specifies a *minimum* version of ios to run (8.0). Resolving that in an 8.0 app (old), will yield GoogleMaps 2.7 (as seen [above](https://github.com/flutter/plugins/pull/2924#issuecomment-672290890)), however new apps that use 9.0 can resolve 3.x versions.

3.9 was good, as seen in the full-green pass linked in the comment above.

3.10 is bad, probably because it exposes new APIs that require a newer version of XCode. The change is [undocumented](https://developers.google.com/maps/documentation/ios-sdk/releases) as of now:

> Version 3.10.0 - August 10, 2020
>
> * Version bump, no changes. This version is functionally identical to v.3.9.0.
> * There is a new Maps Swift Demo App in Google Maps cocoapods.

**Comment by ditman** [comment]
Cleanup ticket created: https://github.com/flutter/flutter/issues/63530

File: packages/google_maps_flutter/google_maps_flutter/ios/google_maps_flutter.podspec [hunk 2]
```diff
@@ -3,7 +3,7 @@
 #
 Pod::Spec.new do |s|
   s.name             = 'google_maps_flutter'
-  s.version          = '0.0.1'
+  s.version          = '0.0.2'
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/ios/google_maps_flutter.podspec:6]: Don't edit this version, it's just a stand in and doesn't mean anything.  `pub` has already handled the versioning solving at the point this podspec is being used.

Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter/ios/google_maps_flutter.podspec]: Done! PTAL!

**Comment by ditman** [comment]
Merged on red; this should make the tree green again.

**Comment by ditman** [comment]
The tree is super-green [now](https://github.com/flutter/plugins/runs/974489019)

**Pull request merged**